pkg/curl/schemes.go

```diff
@@ -35,12 +35,23 @@ var (
 
 // File is a reference to a file fetched through this library.
 type File interface {
-	io.ReaderAt
-
+	fmt.Stringer
 	// URL is the file's original URL.
 	URL() *url.URL
 }
 
+// FileWithCache is a io.ReaderAt with a nice stringer for file's original URL.
+type FileWithCache interface {
+	io.ReaderAt
+	File
+}
+
+// FileWithoutCache is a io.Reader with a nice stringer for file's original URL.
```

Comment:
My understanding is we can submit this temporarily. @hugelgupf is working on a better alternative using memfd.